### ISSUE TITLE
do some validation of the anchor position

### ISSUE THREAD

**Issue by philippotto** [scalableminds/webknossos#6530]

**Comment by normanrz** [comment]
At some point, we will want to implement segment statistics. For that, we need to maintain some bboxes of segments in order to constrain volume calculation, surface calculation, and finding center points (or a point). That will make it feasible to search for new positions if the original ones have been erased.

**Comment by philippotto** [comment]
That's true, but the first two points of this issue are quite an easy fix (I added good-first-issue). So, I wouldn't defer this issue until we got segment statistics.

**Comment by hotzenklotz** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** enhancement
**Summary:** When navigating to a segment's stored anchor position, detect if the voxel at that position no longer contains the expected segment ID and show a helpful recovery UI.

**Current behavior:**
Each segment in a volume annotation stores an `anchorPosition` (a `Vector3` in layer space) in the Redux store. When the user navigates to a segment (e.g., via the segments panel), the app moves the camera to the stored `anchorPosition`. No validation is performed — if the user has since painted a different segment ID over that position, the app silently navigates to a position that no longer belongs to the segment.

**Desired behavior:**
When the app navigates to a segment's anchor position (or when the anchor position is first displayed/used), it should asynchronously check whether the voxel data at that position actually contains the expected segment ID (using the data API). If there is a mismatch:

1. Show an informative UI element (tooltip, notification, or inline warning in the segments panel) that explains:
   - The segment's saved position no longer contains this segment's ID — it was likely overwritten.
   - How to fix it manually: activate the segment ID and brush over the anchor position again.
2. Offer a one-click action to **remove** the segment from the segment list entirely.

The check should be non-blocking (async, no loading spinner required for the main viewport navigation). Only items 1 and 2 above are in scope; searching surrounding data for the segment is explicitly deferred.

**Key interfaces:**
- `Segment` type — the `anchorPosition` field and how it is used in navigation
- `api.data.getDataValue()` — reads the raw (unmapped) segment ID at a given position; already used in the saving/rebasing pipeline
- The segments panel / segment details panel — where the warning UI should appear
- The segment navigation action — where the async check should be triggered

**Acceptance criteria:**
- [ ] When the app navigates to a segment's anchor position and the voxel there does not match the segment ID, an explanatory message is shown (not just a generic error)
- [ ] The message tells the user the position is out of sync and instructs them to brush again with the segment ID to re-anchor it
- [ ] A "Remove segment" action is offered alongside the warning
- [ ] If the anchor position does match the segment ID, no warning is shown
- [ ] If the segment has no anchor position (`null`), the existing behavior is unchanged
- [ ] The check does not block viewport navigation or cause visible lag

**Out of scope:**
- Searching surrounding voxels/bounding box for the segment (item 3 in the original issue — explicitly deferred)
- Segment statistics or bounding box tracking
- Changes to the backend or saving pipeline
